ui/tests/e2e/userpreference.spec.js

```diff
@@ -0,0 +1,77 @@
+import { expect, test } from '@playwright/test';
+import { ENV } from './env';
+
+const userPreferenceTests = [
+  {
+    name: 'Toggle "Meshery Catalog Content"',
+    apiURL: `${ENV.MESHERY_SERVER_URL}/api/user/prefs`,
+    locator: '[data-cy="CatalogContentPreference"]',
+    expectedMethod: 'POST',
+    expectedStatus: 200,
+  },
+  {
+    name: 'Toggle "Send Anonymous Usage Statistics"',
+    apiURL: `${ENV.MESHERY_SERVER_URL}/api/user/prefs?contexts=all`,
+    locator: '[data-cy="UsageStatsPreference"]',
+    expectedMethod: 'POST',
+    expectedStatus: 200,
+  },
+  {
+    name: 'Toggle "Send Anonymous Performance Results"',
+    apiURL: `${ENV.MESHERY_SERVER_URL}/api/user/prefs?contexts=all`,
+    locator: '[data-cy="PerfResultPreference"]',
+    expectedMethod: 'POST',
+    expectedStatus: 200,
+  },
+];
+
+test.describe('User Preferences Page Tests', () => {
+  test.beforeEach(async ({ page }) => {
+    const userPrefReq = page.waitForRequest((request) =>
+      request.url().startsWith(`${ENV.MESHERY_SERVER_URL}/api/user/prefs`),
+    );
+    const userPrefRes = page.waitForResponse((response) =>
+      response.url().startsWith(`${ENV.MESHERY_SERVER_URL}/api/user/prefs`),
+    );
+
+    // Visit User Preferences Page
+    await page.goto(`${ENV.MESHERY_SERVER_URL}/user/preferences`);
+
+    // Verify requests and responses expected on initial page load
+    expect((await userPrefReq).method()).toBe('GET');
+    expect((await userPrefRes).status()).toBe(200);
+
+    // Verify visibility of 'Extensions' Section
+    await expect(
+      page.locator(':nth-child(1) > .MuiFormControl-root > .MuiFormLabel-root'),
```

Comment:
this will break with layout chages

Reply:
Done. Changed to find the elements by their text.